Sources/Logging/Log.swift

```diff
@@ -8,136 +8,137 @@
-            else { throw Error.duplicateDestination(destination.id) }
+            try destinations.modify {
+                guard $0.contains(where: { $0.id == destination.id }) == false else {
+                    throw DefaultLoggerError.duplicateDestination(destination.id)
+                }
 
-            $0.append(destination)
+                $0.append(destination)
+            }
         }
-    }
 
-    public func unregister(_ destination: LogDestination) throws {
+        public func unregisterDestination(_ destination: LogDestination) throws {
 
-        try destinations.modify {
-            guard $0.contains(where: { $0.id == destination.id }) else {
-                throw Error.inexistentDestination(destination.id)
-            }
+            try destinations.modify {
+                guard $0.contains(where: { $0.id == destination.id }) else {
+                    throw DefaultLoggerError.inexistentDestination(destination.id)
+                }
 
-            $0 = $0.filter { $0.id != destination.id }
+                $0 = $0.filter { $0.id != destination.id }
+            }
         }
-    }
 
-    public func removeAllDestinations() {
-
-        destinations.value = []
-    }
+        // MARK: - Module Management
 
-    // MARK: - Logging
+        public func registerModule(_ module: Module, minLevel: Level) throws {
 
-    public func verbose(_ message: @autoclosure () -> String,
-                        file: StaticString = #file,
-                        function: StaticString = #function,
-                        line: UInt = #line) {
+            try modules.modify {
+                guard $0[module] == nil else { throw DefaultLoggerError.duplicateModule(module.rawValue) }
 
-        log(level: .verbose, message: message, file: file, function: function, line: line)
-    }
-
-    public func debug(_ message: @autoclosure () -> String,
-                      file: StaticString = #file,
-                      function: StaticString = #function,
-                      line: UInt = #line) {
+                $0[module] = minLevel
+            }
+        }
 
-        log(level: .debug, message: message, file: file, function: function, line: line)
-    }
+        public func unregisterModule(_ module: Module) throws {
 
-    public func info(_ message: @autoclosure () -> String,
-                     file: StaticString = #file,
-                     function: StaticString = #function,
-                     line: UInt = #line) {
+            try modules.modify {
+                guard let _ = $0.removeValue(forKey: module) else {
+                    throw DefaultLoggerError.inexistentModule(module.rawValue)
+                }
+            }
+        }
 
-        log(level: .info, message: message, file: file, function: function, line: line)
-    }
+        // MARK: - Logging
 
-    public func warning(_ message: @autoclosure () -> String,
+        public func log(module: Module,
```

Comment:
Should we provide a method to log messages without a module?

Reply:
Also thought about it, but then I decided that one could simply have a single module on the app.
It has the downside that we have to use it on every call, though, so I see why it can be cumbersome 😅. One could always make some helper calls in an extension to abstract that away, but it's still not ideal.

That being said, even though I prefer using logging modules across my apps, I can see benefit in continuing to allow the previous (simpler) logging pattern.

Providing this API without a `Module` parameter can be achieved in multiple ways:

1. add another `log` API (requiring no `Module`), performing no module filtering on the `Logger`.
2. modify the existing `log` API to receive a `Module?`, and ignore module filtering on the `Logger` when `nil`.
3. make `LogModule` have a `default` or `none` value, use that on the new `log` API to send to the existing one, and ignore module filtering on the `Logger` for this "special" module.

Option 1 will result in two distinct API's, preserve backwards compatibility (allowing a smooth transition), and avoid extra complexity on the existing `log` implementation (filtering by module).

Options 2 and 3 will still require one to implement a new `log` API without module (internally sending in `nil` or the default value to the ), and will result in more complexity in the `log` implementation (and `LogModule`). This new helper `log` API will also provide backwards compatibility.

Thoughts? Other alternatives?

Reply:
Meanwhile, you went already for the option number 2. Good call and good job. ✨